### PR TITLE
[KOGITO-7922] - Update Kogito Operator fo Go 1.17

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>
See: https://issues.redhat.com/browse/KOGITO-7922

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've ran `make before-pr` and everything is working accordingly
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run operator BDD testing</b>
  Please add comment: <b>/jenkins test</b>

* <b>Run RHPAM operator BDD testing</b>
  Please add comment: <b>/jenkins RHPAM test</b>
</details>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,6 +69,8 @@ GOBIN=$(shell go env GOPATH)/bin
 else
 GOBIN=$(shell go env GOBIN)
 endif
+# set -mod=mod flag to avoid build issues: missing go.sum entry; to add it: ...
```
Comment (radtriste) [inline Makefile:72]: are you sure to force that to all builds ?

**Review by sutaakar — APPROVED** [review]

**Comment by radtriste** [comment]
will setup the label on Jenkins with build from https://github.com/kiegroup/kogito-pipelines/pull/613#issuecomment-1246601369 and then we can try the Jenkins test

**Comment by radtriste** [comment]
@spolti 
```
go get: installing executables with 'go get' in module mode is deprecated.
	To adjust and download dependencies of the current module, use 'go get -d'.
	To install using requirements of the current module, use 'go install'.
	To install ignoring the current module, use 'go install' with a version,
	like 'go install example.com/cmd@latest'.
	For more information, see https://golang.org/doc/go-get-install-deprecation
	or run 'go help get' or 'go help install'.
```

File: Makefile [hunk 2]
```diff
@@ -150,7 +152,7 @@ run: manifests generate fmt vet ## Run a controller from your host.
 
 container-build: ## Build the docker image
 	echo "calling APP container-build ##################################"
-	cekit -v --descriptor kogito-image.yaml build $(BUILDER)
+	cekit -v --no-squash --descriptor kogito-image.yaml build $(BUILDER)
```
Comment (radtriste) [inline Makefile:155]:
```suggestion
	cekit -v --descriptor kogito-image.yaml build $(BUILDER) --no-squash
```

File: Makefile [hunk 3]
```diff
@@ -159,7 +161,7 @@ container-push: ## Push the docker image
 profiling-build: ## Build the profiling docker image
 	cp kogito-image.yaml kogito-image.yaml.save && \
 	cp profiling/image.yaml kogito-image.yaml && \
-	cekit -v --descriptor kogito-image.yaml build $(BUILDER); \
+	cekit -v --no-squash --descriptor kogito-image.yaml build $(BUILDER); \
```
Comment (radtriste) [inline Makefile:164]:
```suggestion
	cekit -v --descriptor kogito-image.yaml build $(BUILDER); --no-squash \
```

File: Makefile [hunk 4]
```diff
@@ -159,7 +161,7 @@ container-push: ## Push the docker image
 profiling-build: ## Build the profiling docker image
 	cp kogito-image.yaml kogito-image.yaml.save && \
 	cp profiling/image.yaml kogito-image.yaml && \
-	cekit -v --descriptor kogito-image.yaml build $(BUILDER); \
+	cekit -v --descriptor kogito-image.yaml build $(BUILDER); --no-squash \
```
Comment (radtriste) [inline Makefile:164]:
```suggestion
	cekit -v --descriptor kogito-image.yaml build $(BUILDER) --no-squash; \
```

File: Makefile [hunk 5]
```diff
@@ -150,7 +152,7 @@ run: manifests generate fmt vet ## Run a controller from your host.
 
 container-build: ## Build the docker image
 	echo "calling APP container-build ##################################"
-	cekit -v --descriptor kogito-image.yaml build $(BUILDER)
+	cekit -v --descriptor kogito-image.yaml build $(BUILDER) --no-squash
```
Comment (radtriste) [inline Makefile:155]:
```suggestion
	cekit -v --descriptor kogito-image.yaml build $(BUILDER)
```

File: Makefile [hunk 6]
```diff
@@ -159,7 +161,7 @@ container-push: ## Push the docker image
 profiling-build: ## Build the profiling docker image
 	cp kogito-image.yaml kogito-image.yaml.save && \
 	cp profiling/image.yaml kogito-image.yaml && \
-	cekit -v --descriptor kogito-image.yaml build $(BUILDER); \
+	cekit -v --descriptor kogito-image.yaml build $(BUILDER) --no-squash; \
```
Comment (radtriste) [inline Makefile:164]:
```suggestion
	cekit -v --descriptor kogito-image.yaml build $(BUILDER); \
```

**Comment by radtriste** [comment]
@spolti I updated `kustomize` (to latest version) and `operator-sdk` (to latest golang 1.17 supported version)

**Comment by spolti** [comment]
> @spolti I updated `kustomize` (to latest version) and `operator-sdk` (to latest golang 1.17 supported version)

Didn't the operator-sdk upgrade require changes in the code base?

**Comment by radtriste** [comment]
does not seems so :D

**Pull request merged**